src/ai/ai_core.cpp

```diff
@@ -111,11 +116,12 @@
 	delete c->ai_instance;
 	c->ai_instance = nullptr;
 	c->ai_info = nullptr;
+	delete c->ai_config;
+	c->ai_config = nullptr;
```

Comment:
As I hinted earlier on Discord, maybe a unique_ptr makes this all a bit more pretty, as then you just have to reset the unique_ptr and not manage anything else :)

Reply:
Yeah, I tried, but incomplete type is a pain.

Reply:
Found how to fix my unique_ptr issue :)